server/webhook.go

```diff
@@ -0,0 +1,213 @@
+package main
+
+import (
+	"crypto/hmac"
+	"crypto/sha256"
+	"crypto/subtle"
+	"encoding/hex"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"math"
+	"net/http"
+	"strings"
+	"time"
+
+	"github.com/mattermost/mattermost-server/v6/model"
+	"github.com/pkg/errors"
+
+	"github.com/mattermost/mattermost-plugin-zoom/server/zoom"
+)
+
+func (p *Plugin) handleWebhook(w http.ResponseWriter, r *http.Request) {
+	if !p.verifyWebhookSecret(r) {
+		p.API.LogWarn("Could not verify webhook secret")
+		http.Error(w, "Not authorized", http.StatusUnauthorized)
+		return
+	}
+
+	if !strings.Contains(r.Header.Get("Content-Type"), "application/json") {
+		res := fmt.Sprintf("Expected Content-Type 'application/json' for webhook request, received '%s'.", r.Header.Get("Content-Type"))
+		p.API.LogWarn(res)
+		http.Error(w, res, http.StatusBadRequest)
+		return
+	}
+
+	b, err := ioutil.ReadAll(r.Body)
+	if err != nil {
+		p.API.LogWarn("Cannot read body from Webhook")
+		http.Error(w, err.Error(), http.StatusBadRequest)
+		return
+	}
+
+	var webhook zoom.Webhook
+	if err = json.Unmarshal(b, &webhook); err != nil {
+		p.API.LogError("Error unmarshaling webhook", "err", err.Error())
+		http.Error(w, err.Error(), http.StatusBadRequest)
+		return
+	}
```

Comment:
I think this is written the way that it is because we would essentially be "discarding" the HTTP body that would need to be re-read if `json.NewDecoder` was used.